py/visdom/__init__.py

```diff
@@ -2312,3 +2318,115 @@ def dual_axis_lines(self, X=None, Y1=None, Y2=None, opts=None, win=None, env=Non
+                        each dict.
+        """        
+        try:
+            from cerberus import Validator
+        except:
+            raise RuntimeError(
+                "Cerebus must be installed to plot Graph figures")
+
+        # Raise error if type not among one of the types
+        assert opts.get("type", "gph") in [
+            "gph", "dgph", "wgph", "dwgph"],\
+            "Invalid graph type {}, should be one of ['gph', 'dgph', 'wgph', 'dwgph']".format(
+                opts.get("type"))
+
+        # Raise error if labels is other than custom and default
+        assert opts.get("labels", "default") in ["default", "custom"],\
+            "Invalid labels: {}, labels should be one of ['default', 'custom']".format(
+                opts.get("labels"))
+
+        # validation of user input Dict
+        schema_for_graph = {
+            "value": {'type': 'integer', 'required': True},
+            "weight": {'type': 'integer', "required": True if opts.get('type', "gph") in ["wgph", "dwgph"] else False},
+            "label": {'type': 'string', "required": True if opts.get('labels', "default") == "custom" else False}
+        }
+        v = Validator(schema_for_graph)
+        for key, value in graphDict.items():
+            for val in value:
+                if not v.validate(val):
+                    raise RuntimeError(v.errors)
+
+        # Creating Data in defined format
+        links = []
+        for key, value in graphDict.items():
+            for v in value:
+                link = {}
+                link["source"] = key
+                link["target"] = v["value"]
+                link["weight"] = v.get("weight", 0)
+                if opts.get("labels") == "custom":
+                    link["label"] = v["label"]
+                elif opts.get("labels") == "default":
+                    if opts.get("type", "gph") in ["dwgph", "wgph"]:
+                        link["label"] = v["weight"]
+                    else:
+                        link["label"] = str(key) + "-" + str(v["value"])
+                links.append(link)
+
+        nodes = []
+        visited = set()
+        for link in links:
+            if link["source"] not in visited:
+                node = {"name": str(link["source"])}
+                nodes.append(node)
+                visited.add(link["source"])
+            if link["target"] not in visited:
+                node = {"name": str(link["target"])}
+                nodes.append(node)
+                visited.add(link["target"])
+
+        for i in range(len(visited)):
+            if i not in visited:
+                raise RuntimeError(
+                    "Graph you provided should contain nodes from 0 to n-1, but {} was missing.".format(i))
+
+        GData = {
+            "nodes": nodes,
+            "links": links
+        }
+
+        jsonstring = json.dumps(GData)
+        GData = json.loads(jsonstring)
```

Comment:
Any reason for the dump and reload here? I tend to prefer comments for the rationale on things that are 'no-ops' outside of externalities.

Reply:
Right. I think this part was done for debugging and therefore could and should be removed.
Or did it have any other purpose?

Reply:
Nicely pointed out @JackUrb  and @dizcza. It was done for debugging purpose only. I will make the changes. 
Thanks